addons/l10n_it_stock_ddt/report/l10n_it_ddt_report.xml

```diff
@@ -168,6 +168,13 @@
             </t>
         </t>
     </template>
+
+    <template id="report_ddt_view">
+        <t t-foreach="docs" t-as="o">
+            <t t-call="l10n_it_stock_ddt.report_ddt" t-lang="o.partner_id.lang"/>
```

Comment:
This should only be needed for the product names, but I don't see why you would not do it in the report itself as in https://github.com/odoo/odoo/blob/9b9c4a3433d2a052e70e434b11817b791f38e508/addons/stock/report/report_deliveryslip.xml

Reply:
It translates the document fields whereas `<t t-set="o" t-value="o.with_context(lang=o.partner_id.lang)" />` translates the product names. I think both are necessary so all the document is in the partner's language

Reply:
to expand on jco answer: `<t t-set="o" t-value="o.with_context(lang=o.partner_id.lang)" />` is so that calls to o.[something].[something] so odoo fields values are translated.

`<t t-call="..." t-lang="o.partner_id.lang"/>` is so that the template itself is translated (so string appearing in the template itself)

so it depends on what we want to have translated

Reply:
The report is generated in English when there is no partner_id. Should I check if a partner_id is provided and in the case where there is not, translate the report in another language ?